apps/io-func-wallet-solution-backend/src/infra/pdv-tokenizer/client.ts

```diff
@@ -0,0 +1,121 @@
+import { FiscalCode, NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import * as E from "fp-ts/lib/Either";
+import * as TE from "fp-ts/lib/TaskEither";
+import { flow, pipe } from "fp-ts/function";
+import * as t from "io-ts";
+import { v4 as uuidv4 } from "uuid";
+import { PdvTokenizerHealthCheck } from "./health-check";
+import { UserIdRepository } from "@/user";
+import { PdvTokenizerApiClientConfig } from "@/app/config";
+
+const Token = t.type({
+  token: NonEmptyString,
+});
+
+const Pii = t.type({
+  pii: FiscalCode,
+});
+
+export class PdvTokenizerClient
+  implements UserIdRepository, PdvTokenizerHealthCheck
+{
+  #baseURL: string;
+  #options: RequestInit;
+
+  constructor({ baseURL, apiKey }: PdvTokenizerApiClientConfig) {
+    this.#baseURL = baseURL;
+    this.#options = {
+      headers: {
+        "x-api-key": apiKey,
+        Accept: "application/json, text/plain",
+        "Content-Type": "application/json",
+      },
+    };
+  }
+
+  getUserIdByFiscalCode = (fiscalCode: FiscalCode) =>
+    pipe(
+      TE.tryCatch(
+        async () => {
+          const result = await fetch(
+            `${this.#baseURL}/tokenizer/v1/tokens/search`,
+            {
+              ...this.#options,
+              signal: AbortSignal.timeout(3000),
+              method: "POST",
+              body: JSON.stringify({
+                pii: fiscalCode,
+              }),
+            }
+          );
+          if (!result.ok) {
+            throw new Error(JSON.stringify(await result.json()));
+          }
+          return result.json();
+        },
+        (error) => new Error(`error getting user id by fiscal code: ${error}`)
+      ),
+      TE.chain(
+        flow(
+          Token.decode,
+          E.mapLeft(
+            () =>
+              new Error(
+                "error getting user id by fiscal code: invalid result format from pdv"
+              )
+          ),
+          TE.fromEither
+        )
+      ),
+      TE.map(({ token }) => ({ id: token }))
+    );
+
+  getFiscalCodeByUserId = (id: string) =>
+    pipe(
+      TE.tryCatch(
+        async () => {
+          const result = await this.getFiscalCode(id);
+          if (!result.ok) {
+            throw new Error(JSON.stringify(await result.json()));
+          }
+          return result.json();
+        },
+        (error) => new Error(`error getting fiscal code by user id: ${error}`)
+      ),
+      TE.chain(
+        flow(
+          Pii.decode,
+          E.mapLeft(
+            () =>
+              new Error(
+                "error getting fiscal code id by user id: invalid result format from pdv"
+              )
+          ),
+          TE.fromEither
+        )
+      ),
+      TE.map(({ pii }) => ({
+        fiscalCode: pii,
+      }))
+    );
+
+  healthCheck = () =>
+    pipe(
+      TE.tryCatch(
+        async () => {
+          const uuid = uuidv4();
+          const result = await this.getFiscalCode(uuid);
```

Comment:
It seems that this healthcheck using a random `uuid` may always return an error. What do you think about using a real uuid? Or a test one if it exists. In this way we will be sure that it is really working and not just that the endpoint is reachable.

Reply:
I asked about a test uuid

Reply:
they are gonna give us a test CF. Maybe we should close this PR and create a new jira task since it might take a while

Reply:
they will provide us with the API key and then we can generate a test CF/token. Maybe we should close this PR and create a new jira task since it might take a while

Reply:
Ok, for now I would therefore say to put a mocked uuid, perhaps read from the env. what do you think?